### PR TITLE
Add github action to update instance-types

### PR DESCRIPTION
**Pull request by agrare**
Example run: https://github.com/agrare/manageiq-providers-amazon/actions/runs/1993106302
Example PR: https://github.com/agrare/manageiq-providers-amazon/pull/2, https://github.com/agrare/manageiq-providers-amazon/pull/5

TODO:
- [x] Drop postgres service requirement
- [x] Use a personal access token so that tests can run on the created PR

### CODE REVIEW

**Comment by chessbyte** [comment]
This is very cool!  Will it only create a PR if there are changes?

**Comment by agrare** [comment]
> Will it only create a PR if there are changes?

Correct, per https://github.com/marketplace/actions/create-pull-request#action-behaviour "If there are no changes (i.e. no diff exists with the checked-out base branch), no pull request will be created and the action exits silently."

File: .github/workflows/instance_types.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up system
+      run: bin/before_install
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 2.7
+        bundler-cache: true
+    - name: Bundle
+      run: bundle update
```
Comment (Fryguy) [inline .github/workflows/instance_types.yaml:20]: I don't even think this is needed because setup-ruby should handle it

**Comment by chessbyte** [comment]
Why is `db/fixtures/aws_instance_types.yml` being changed in this PR?  I had expected that to happen as a result of this PR being merged and operational.

**Comment by agrare** [comment]
@chessbyte because I am testing that with no changes this doesn't create a pull request

File: .gitignore [hunk 2]
```diff
@@ -7,6 +7,7 @@
 /pkg/
 /spec/reports/
 /tmp/
+/vendor/bundle
```
[the remark this answers is not in the document]
Comment (agrare) [reply .gitignore]: https://github.com/peter-evans/create-pull-request/issues/1095

Latest `@v4` resolves this issue

**Comment by agrare** [comment]
@Fryguy if you're good with this approach do you want to continue to pursue the "Use a personal access token so that tests can run on the created PR" so github actions will run on PRs created by this action?

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (Fryguy) [reply .gitignore:10]: This gitignore change is no longer needed, then, right?

**Comment by Fryguy** [comment]
Yes, I'm good with this approach.

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (agrare) [reply .gitignore:10]: Technically no, but we do `.gitignore` it in core and don't see a downside other than consistency with the other plugins (nothing a pr-blaster couldn't fix)

**Comment by Fryguy** [comment]
I created the token but where do I put it?  Is the built-in GITHUB_TOKEN not sufficient?

**Comment by agrare** [comment]
If you add it to the repo secrets (https://github.com/ManageIQ/manageiq-providers-amazon/settings/secrets/actions) I can reference it from there by name

**Comment by Fryguy** [comment]
I think you can use the built-in one though, if I'm not mistaken.  I understand the limitation described in the create-pull-request action, but I interpreted that to mean that once we merge, the next time it runs it will potentially create a commit that, in turn, will _not_ run the push workflow, which is fine since we have daily crons anyway.

**Comment by Fryguy** [comment]
ok it's `DEPLOY_TOKEN`.  As discussed, let's also add `workflow_dispatch` to the `on` section to allow manual runs.

**Review by Fryguy — APPROVED** [review]

**Comment by agrare** [comment]
With `on: workflow_dispatch` we have the option of manually running the action:
image

**Comment by Fryguy** [comment]
That is SO COOL! 😍

**Comment by agrare** [comment]
Okay tested running a manual workflow on my fork, creates a PR and tests are running: https://github.com/agrare/manageiq-providers-amazon/pull/5

Taking out of WIP

File: .github/workflows/instance_types.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        ruby-version: 2.7
+        bundler-cache: true
+    - name: Run instance_types rake task
+      run: bundle exec rake aws:extract:instance_types
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v4
+      with:
+        add-paths: db/fixtures/aws_instance_types.yml
+        commit-message: Update AWS instance_types
+        title: Update AWS instance_types
```
Comment (Fryguy) [inline .github/workflows/instance_types.yaml:27]: Maybe for a follow-up , but I'm think we should set `body:` here as well.  The default one isn't very informative.

Comment (Fryguy) [reply .github/workflows/instance_types.yaml:27]: I'd be fine with leaving blank as the title is informative enough, or perhaps we could add in the date?

**Review by Fryguy — APPROVED** [review]

**Pull request merged**